external_chi2/optimal_observables/interface_oos.py

```diff
@@ -27,16 +28,18 @@ def __init__(self, coefficients):
             '{collider}_ww_semilep_365': 'invcov_{collider}_ww_semilep_365.dat'
         }
 
-        self.invcovs = [np.loadtxt(current_file_path / path.format(collider=collider)) for path in self.datasets.values()]
+        incovs_reordered = []
+        for path in self.datasets.values():
+            invcov = np.loadtxt(current_file_path / path.format(collider=collider))
+            temp = self.project.T @ invcov @ self.project
+            incovs_reordered.append(temp)
+        self.incovs_reordered = jnp.sum(jnp.array(incovs_reordered), axis=0)
 
         self.n_dat = len(oo_wc_basis)
 
     def compute_chi2(self, coefficient_values):
 
-        chi2_value = 0
-        for invcov in self.invcovs:
-            chi2_value += np.linalg.multi_dot(
-                [coefficient_values, self.project.T, invcov, self.project, coefficient_values])
+        chi2_value = coefficient_values @ self.incovs_reordered @ coefficient_values
```

Comment:
```suggestion
        chi2_value = jnp.einsum("i, ij, j", coefficient_values, self.invcov_tot, coefficient_values)
```

It should be more efficient than @